task stage missing for Gradle compile on Heroku

<!--
- Please follow the issue template below for bug reports.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-->

##### **Overview of the issue**

If using Gradle, running `jhipster heroku` and choosing `Git (compile on Heroku)` then error occurs and app is not deployed to Heroku.
```
remote:        FAILURE: Build failed with an exception.
remote:
remote:        * What went wrong:
remote:        Task 'stage' not found in root project 'jhipster'.
remote:
remote:        * Try:
remote:        Run gradlew tasks to get a list of available tasks. Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
remote:
remote:        * Get more help at https://help.gradle.org
remote:
remote:        BUILD FAILED in 23s
remote:
remote:  !     ERROR: Failed to run Gradle!
remote:        It looks like your project does not contain a 'stage' task, which Heroku needs in order
remote:        to build your app. Our Dev Center article on preparing a Gradle application for Heroku
remote:        describes how to create this task:
remote:        https://devcenter.heroku.com/articles/deploying-gradle-apps-on-heroku
remote:
remote:        If you're stilling having trouble, please submit a ticket so we can help:
remote:        https://help.heroku.com
remote:
remote:        Thanks,
remote:        Heroku
remote:
remote:  !     Push rejected, failed to compile Gradle app.
remote:
remote:  !     Push failed
remote: Verifying deploy...
remote:
remote: !       Push rejected to jhipster-gradle-git-error.
```
<!-- Explain the bug, if an error is being thrown a stack trace helps -->

##### **Motivation for or Use Case**

<!-- Explain why this is a bug for you -->

##### **Reproduce the error**

<!-- For bug reports, an unambiguous set of steps to reproduce the error -->

* generate project choosing **Gradle**
* run `jhipster heroku` choosing **Git (compile on Heroku)**

##### **Related issues**

<!-- Has a similar issue been reported before? Please search both closed & open issues -->

##### **Suggest a Fix**

<!-- If you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit) -->

This problem exists since https://github.com/jhipster/generator-jhipster/commit/e0ff9dee

Solution: add back task `stage` with some modifications:
* put it into https://github.com/jhipster/generator-jhipster/blob/master/generators/heroku/templates/heroku.gradle.ejs - this keeps Heroku related code separated and prevents accidental removal in the future
* use `bootJar` instead of `bootWar`:
```
task stage(dependsOn: 'bootJar') {
}
```

##### **JHipster Version(s)**

```
jhipster@0.0.0 c:\projects\teest\jhipster-heroku-error
`-- generator-jhipster@6.0.1

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.mycompany.myapp",
      "nativeLanguage": "et"
    },
    "jhipsterVersion": "6.0.1",
    "applicationType": "monolith",
    "baseName": "jhipster",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "gradle",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "bXktc2VjcmV0LXRva2VuLXRvLWNoYW5nZS1pbi1wcm9kdWN0aW9uLWFuZC10by1rZWVwLWluLWEtc2VjdXJlLXBsYWNl",
    "clientFramework": "angularX",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "useSass": true,
    "clientPackageManager": "npm",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "enableTranslation": true,
    "nativeLanguage": "et",
    "languages": ["et", "en", "ru"],
    "herokuAppName": "jhipster-gradle-git-error",
    "herokuDeployType": "git"
  }
}

</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>


##### **Environment and Tools**

java version "1.8.0_211"
Java(TM) SE Runtime Environment (build 1.8.0_211-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.211-b12, mixed mode)

git version 2.21.0.windows.1

node: v10.15.3

npm: 6.9.0

yarn: 1.15.2

Docker version 18.09.2, build 6247962

docker-compose version 1.23.2, build 1110ad01

##### **Browsers and Operating System**

<!-- What OS are you on? is this a problem with all browsers or only IE8? -->

Windows 10

-   [x] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

--- a/generators/heroku/templates/heroku.gradle.ejs
+++ b/generators/heroku/templates/heroku.gradle.ejs
@@ -22,6 +22,13 @@ heroku {
   appName = "<%= herokuAppName %>"
   buildpacks = ["heroku/jvm"]
 }
+<%_ if (herokuDeployType === 'git') { _%>
+
+// Task stage is used by Heroku, see also
+// https://devcenter.heroku.com/articles/deploying-gradle-apps-on-heroku
+// and GRADLE_TASK configuration variable.
+task stage(dependsOn: "bootJar") {
+}
 
 gradle.taskGraph.whenReady {taskGraph ->
     taskGraph.afterTask() {task ->
@@ -31,3 +38,4 @@ gradle.taskGraph.whenReady {taskGraph ->
         }
     }
 }
+<%_ } _%>

